File: DSA/Greedy/Problems/6.N Meetings in One Room/sol.py
class Solution:
    def maxMeetings(self, start, end, n):
        meetings = sorted(zip(end, start))
        count = 1
        last_end = meetings[0][0]

        for i in range(1, n):
            if meetings[i][1] > last_end:
                count += 1
                last_end = meetings[i][0]
        return count

File: DSA/Greedy/Problems/6.N Meetings in One Room/sol.cpp
class Solution {
public:
    int maxMeetings(int start[], int end[], int n) {
        vector<pair<int, int>> meetings;
        for (int i = 0; i < n; i++)
            meetings.push_back({end[i], start[i]});
        sort(meetings.begin(), meetings.end());

        int count = 1, last_end = meetings[0].first;
        for (int i = 1; i < n; i++) {
            if (meetings[i].second > last_end) {
                count++;
                last_end = meetings[i].first;
            }
        }
        return count;
    }
};
